Add recently played games sensor to Xbox integration

AmGarera opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->
N/A - This is a new feature, not a breaking change.

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This PR adds a new "Recently Played Games" sensor to the Xbox integration that tracks recently played games with comprehensive achievement progress and metadata.

**Key features:**
- Displays count of recently played games (up to 10) as the sensor state
- Rich attributes for each game including:
  - Game title and title ID
  - Last played timestamp
  - Achievement progress (as integer 0-100 for automation compatibility)
  - Achievements earned/total
  - Gamerscore earned/total
  - Image URL (Poster or Logo) when available
- State persistence using `RestoreSensor` - sensor shows cached data immediately after restart (before 5-minute coordinator update)
- Measurement state class for statistics support
- Automatic updates every 5 minutes via `XboxTitleHistoryCoordinator`

**Implementation:**
- Follows existing coordinator pattern used by `XboxConsolesCoordinator`
- Defensive None checks for optional fields
- Full translation support in strings.json
- Comprehensive unit tests with 100% coverage

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/43147
- Link to developer documentation pull request: 
- Link to frontend pull request: 

**Technical details:**
- Added `XboxTitleHistoryCoordinator` that fetches title history with `TitleFields.ACHIEVEMENT` and `TitleFields.IMAGE`
- Added `XboxTitleHistorySensorEntity` with `RestoreSensor` for state persistence
- Added `XboxTitleHistorySensorEntityDescription` for entity configuration
- Achievement progress stored as integer (0-100) instead of string for better automation/template support
- Image URLs prefer Poster type, fall back to Logo type
- Game name defaults to "Unknown", title_id defaults to empty string for robustness

**Changes made:**
- `homeassistant/components/xbox/sensor.py`: Added sensor entity, description, and attributes function
- `homeassistant/components/xbox/coordinator.py`: Added title history coordinator with 5-minute update interval
- `homeassistant/components/xbox/strings.json`: Added translations for sensor and attributes
- `tests/components/xbox/test_sensor.py`: Added comprehensive tests for sensor with/without data


## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.

  AI tools are welcome, but contributors are responsible for *fully*
  understanding the code before submitting a PR.
-->

- [x] I understand the code I am submitting and can explain how it works.
- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.
- [x] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

AmGarera committed:
Add recently played games sensor to Xbox integration

Implement a new sensor that displays recently played games with
achievement progress tracking. The sensor shows the count of recently
played games as its state, with detailed game information including
achievement progress, gamerscore, and last played time in attributes.

Features:
- New XboxTitleHistoryCoordinator that fetches title history every 5 minutes
- Recently played games sensor showing up to 10 games
- Per-game achievement progress tracking (earned/total achievements and gamerscore)
- Last played timestamp for each game
- Full translation support in strings.json
- Comprehensive unit tests including edge cases

The implementation follows the existing coordinator pattern used by
XboxConsolesCoordinator, keeping the architecture consistent and
maintainable. All code passes ruff linting and includes proper error
handling and null safety checks.

Tests verify:
- Sensor creation and state reporting
- Game attributes with achievement data
- Graceful handling of empty game history
- Proper entity registry integration

AmGarera committed:
Add state restoration to Xbox recently played games sensor

AmGarera committed:
Address code review feedback for Xbox recently played games sensor

- Add image_url to game attributes for visual presentation
- Change achievement_progress from string to int for better automation support
- Add defensive None checks for game.name and game.title_id
- Implement proper state restoration with type conversion
- Add state_class=MEASUREMENT for statistics support
- Update tests with coordinator refresh and fix assertions
- Update test snapshots for new sensor

AmGarera committed:
Merge branch 'dev' into features/xbox-recently-played

AmGarera committed:
Merge branch 'dev' into features/xbox-recently-played

joostlek commented on homeassistant/components/xbox/coordinator.py:
```diff
@@ -344,3 +345,49 @@ async def _async_update_data(self) -> SmartglassConsoleList:
             ) from e
+
+
+class XboxTitleHistoryCoordinator(DataUpdateCoordinator[TitleHubResponse | None]):
```
Why would the type be None?

AmGarera replied:
Removed it, it can never be None

joostlek commented on homeassistant/components/xbox/sensor.py:
```diff
@@ -359,3 +441,58 @@ def native_value(self) -> StateType:
+                return self.entity_description.attributes_fn(self.coordinator.data)
+            # Fall back to restored attributes until first update
+            return self._restored_attributes
+        return None
```
Can we exclude the extra state attributes? Ideally we don't introduce big objects in them, or things that can be contextualized. Ideally it becomes a service, just like we have in overseerr, then users can make their own sensors using template triggers

AmGarera committed:
Address code review feedback for Xbox recently played games sensor

- Add type: ignore[misc] for AuthenticationManager inheritance
- Add type: ignore[no-any-return] annotations for datetime return types
- Remove optional None from TitleHubResponse type as it's never None
- Fix strings.json formatting (unit_of_measurement placement)

AmGarera committed:
Replace Xbox recently played games attributes with service

AmGarera committed:
Merge dev branch changes into Xbox recently played games feature

- Integrate async_migrate_entry with friends subentries migration
- Keep XboxTitleHistoryCoordinator for recently played games service
- Add async_setup() for service registration
- Update imports to include new dev dependencies
- Keep all recently played games functionality intact

AmGarera committed:
Add polish and optimize Xbox recently played games feature

- Add account context (xuid, account_name) to service response
- Document service return schema in strings.json
- Add usage guidance in service docstring
- Optimize title history updates with game-change detection:
  * Listen to real-time status updates (every 10 seconds)
  * Only fetch title history when game actually changes
  * Ignore system apps (Home/Settings/Store) via is_game filter
  * Keep 30-minute fallback for offline periods
  * Reduces API calls from 12/hour to 2-4/hour typical usage

joostlek reviewed (changes requested):
there's a merge conflict, can you take a look?
